packages/angular/build/src/tools/esbuild/loader-import-attribute-plugin.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import type { Loader, Plugin } from 'esbuild';
+import { readFile } from 'node:fs/promises';
+
+const SUPPORTED_LOADERS: Loader[] = ['binary', 'file', 'text'];
+
+export function createLoaderImportAttributePlugin(): Plugin {
+  return {
+    name: 'angular-loader-import-attributes',
+    setup(build) {
+      build.onLoad({ filter: /./ }, async (args) => {
+        const loader = args.with['loader'] as Loader | undefined;
```

Comment:
I wonder if there's a feature request for esbuild to have the ability to configure a plugin filter based on an import attribute.

Reply:
Discussed here: https://github.com/evanw/esbuild/issues/3612#issuecomment-1912486781
So it's possible it will be available in the future and we can then transition to that when ready.